### PR TITLE
CS-3005-solo

### PR DESCRIPTION
**Pull request by sdorleTVPage**
Please review the solo code on master also, as it has most of the code.

### CODE REVIEW

File: layouts/solo/main-ajax.html [hunk 1]
```diff
@@ -40,16 +40,15 @@
...
-          value: "coffee",
-          label: "main_category"
-        }]
-      }
+      channelId : "84956291",
+      attributes: {             // All the attributes will go here. Put referenceId as comma seperated for multiple
+        referenceId: "02,03",
+        category_id_1: "123",
+        category_id_2: "123"
+      }
```
Comment (rfornes) [inline layouts/solo/main-ajax.html]: This shall be like this:

` attributes: {
          search: "### REPLACE WITH SEARCH QUERY",
          
          brand: ["### REPLACE WITH BRAND(S) ###"],
          
          category: [{
            value: "### REPLACE WITH CATEGORY ID ###",
            label: "### REPLACE WITH CATEGORY NAME ###"
          }],
       }`

This will be defined soon here https://tvpage.atlassian.net/wiki/display/DEV/Widget+Embed+Standards

File: static/solo/lib.min.js [hunk 2]
```diff
@@ -10745,14 +10745,13 @@ define('jquery-private',['vendor/jquery'],function(){
 define('src/js/player/settings',{
     controls: {
         active: true,
-        seekBar: { progressColor: '#273691' },
-        floater: { removeControls: ['tvplogo', 'hd'], transcript: false }
+        seekBar: { progressColor: __TVPage__.config[0].settings.progresscolor },
+        floater: { removeControls:  __TVPage__.config[0].settings.removecontrols, transcript:  __TVPage__.config[0].settings.transcript  }
```
Comment (rfornes) [inline static/solo/lib.min.js]: No checks? this will brake if no settings are passed

File: static/solo/lib.min.js [hunk 3]
```diff
@@ -10880,28 +10881,18 @@ define('src/js/index',['require','jquery-private','./player/index'],function(req
...
   })
-
-  // $.ajax({
-  //   url: location.protocol + "//app.tvpage.com/api/videos/search",
-  //   dataType: 'jsonp',
-  //   data: {'X-login-id': config.loginid, ''+config.attributes.category[0].value+'': config.attributes.category[0].value }
-  // }).done(function(res){
-  //   console.log(res);
-  //   player.init(res,function(){});
-  // })
```
Comment (rfornes) [inline static/solo/lib.min.js]: ?

File: static/solo/src/js/player/index.js [hunk 4]
```diff
@@ -86,6 +87,7 @@ define(function(require) {
           player.on('tvp:media:videoended', handleEnded);
           player.on('tvp:media:ready', function(){
             playerReady = true;
+            $('#'+player.options.DOMContainer.id).find('.tvp-progress-bar').css('background-color', settings.controls.seekBar.progressColor);
```
Comment (rfornes) [inline static/solo/src/js/player/index.js]: No checks?

File: static/solo/src/js/player/index.js [hunk 5]
```diff
@@ -86,14 +87,53 @@ define(function(require) {
...
               callback();
             }
           });
           resize();
         };
-        var defaults = require('./settings');
-        var settings = $.extend( {}, defaults, {divId:targetId });
+
+        // Check for settings is passed or not
+        if(config && config.settings){
```
Comment (rfornes) [inline static/solo/src/js/player/index.js]: Did you tested this?
If `settings` does not exists this will brake.
One way to check for an existing property is: `if ("undefined" !== typeof config.settings)`

File: static/solo/src/js/player/index.js [hunk 6]
```diff
@@ -86,14 +87,53 @@ define(function(require) {
...
+          if(config.settings.hasOwnProperty('progresscolor')){
+            var progresscolor = config.settings.progresscolor;
+          }else{
+            var progresscolor = '#E57211';
+          }
```
Comment (rfornes) [inline static/solo/src/js/player/index.js]: You are declaring a variable twice, this is not good practice.
You can declare it first and then conditionally set it's value.

Comment (rfornes) [inline static/solo/src/js/player/index.js]: Ah you are declaring it thrice actually, @sdorleTVPage are you looking at the Scrutinizr feedback? code didn'tpassed quality check

File: static/solo/lib.min.js (hunk 3 above)
Comment (sdorleTVPage) [inline static/solo/lib.min.js]: removed the unwanted stuff

File: layouts/solo/main-a.html [hunk 7]
```diff
@@ -31,7 +31,7 @@
   var s, as = d.getElementsByTagName(t)[0];
   if (d.getElementById(id)) return;
   s = d.createElement(t); s.id = id;
-  s.src = '//d2kmhr1caomykv.cloudfront.net/player/assets/tvp/tvp-{{ ($.Scratch.Get "settings").version }}-min.js';
+  s.src = "http://localhost:1313/solo/player-debug.js";
```
Comment (liorkuyer) [inline layouts/solo/main-a.html]: that doesn't seep right

File: layouts/solo/main-c.html [hunk 8]
```diff
@@ -40,15 +40,14 @@
...
       id: "tvpembed-{{ .File.BaseFileName }}",
-      settings: {{ $.Scratch.Get "settings" }},
-      attributes: {
-        referenceIds: ["02","01"],
-        category: [{
-          value: "coffee",
-          label: "main_category"
-        }]
+      channelId : "84956291",
+      parameters: {
```
Comment (liorkuyer) [inline layouts/solo/main-c.html]: This goes under "channel" key. 

channel: {
 id: '',
"parameters"
}

File: static/solo/lib.min.js [hunk 9]
```diff
@@ -10850,11 +10843,37 @@ define('src/js/player/index',['require','jquery-private','./settings'],function(
...
         };
-        var defaults = require('./settings');
-        var settings = $.extend( {}, defaults, {divId:targetId });
+
+        // Check for settings is passed or not
+        if(config && "undefined" !== typeof config.settings){
+          if(config.settings.hasOwnProperty('progresscolor')){
+            progresscolor = config.settings.progresscolor;
+          }
+          if(config.settings.hasOwnProperty('removecontrols')){
```
Comment (liorkuyer) [inline static/solo/lib.min.js]: Move player settings to "playback". nothing is stored directly on settings

File: static/solo/src/js/player/index.js [hunk 10]
```diff
@@ -92,11 +99,37 @@ define(function(require) {
...
+          poster: true,
+          techOrder: 'html5,flash',
+          analytics: { tvpa: false },
+          apiBaseUrl: '//app.tvpage.com',
+          swf: "//d2kmhr1caomykv.cloudfront.net/player/assets/tvp/tvp-1.8.3-flash.swf"
+        };
 
         (function poller( ){
-          var deferred = setTimeout(function(){
+          setTimeout(function(){
```
Comment (liorkuyer) [inline static/solo/src/js/player/index.js]: why is settimeout needed?

File: layouts/solo/main-a.html (hunk 7 above)
Comment (sdorleTVPage) [inline layouts/solo/main-a.html]: Yes, Ricky is going to fix the player lib and make a new version.
Until that we put the fix into a player-debug.js

File: layouts/solo/main-c.html (hunk 8 above)
Comment (sdorleTVPage) [inline layouts/solo/main-c.html]: will do

File: static/solo/lib.min.js (hunk 9 above)
Comment (sdorleTVPage) [inline static/solo/lib.min.js]: will do

File: static/solo/src/js/player/index.js (hunk 10 above)
Comment (sdorleTVPage) [inline static/solo/src/js/player/index.js]: sometimes it takes time to initialize the player, so we are polling
